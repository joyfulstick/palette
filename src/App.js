import React, { Component } from 'react'
import Palette from './containers/Palette/Palette'

class App extends Component {
  render() {
    return (
      <div className="App">
        <Palette />
      </div>
    )
  }
}

export default App
