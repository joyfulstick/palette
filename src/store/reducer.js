const initialState = {
  value: 1,
  radius:
    window.innerWidth > window.innerHeight
      ? Math.round(window.innerWidth / 6)
      : Math.round(window.innerHeight / 6),
  rgbColors: {
    rgb: '',
    hex: '',
  },
  hsl: '',
  alpha: 0,
  schemeModel: 1,
  schemes: [],
  picking: true,
  css: '',
  inputValue: '',
}

const reducer = (state = initialState, action) => {
  // switch (action.type) {
  //   case '':
  //     return
  //   default:
  //     state
  // }
  return state
}

export default reducer
