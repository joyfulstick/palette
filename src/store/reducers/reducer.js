import * as actionTypes from '../actions/actionTypes'
import {
  arrayToRgbString,
  hexToRgb,
  rgbStringToHex,
  rgbToHex,
  schemesGenerator,
  updatedObject,
} from '../../lib/utilities'

import { COLOR_PATTERN } from '../../constants/patterns'

const initialState = {
  value: 1,
  rgbColors: {
    rgb: 'rgb(72,160,255)',
    hex: '#48a0ff',
    r: 72,
    g: 160,
    b: 255,
  },
  hsl: '',
  alpha: 0,
  schemeModel: 2,
  schemes: [
    ['#48a0ff', '#ffa748'],
    ['#004fa3', '#a35500'],
    ['#00346d', '#6d3900'],
    ['#002752', '#522a00'],
    ['#001f41', '#412200'],
  ],
  picking: true,
}

const valueControl = (state, action) => {
  action.payload.event.persist()
  let { value } = action.payload.event.target
  value = +value
  const { r, g, b } = state.rgbColors
  const schemes = schemesGenerator(
    r * value,
    g * value,
    b * value,
    state.schemeModel,
  )
  return updatedObject(state, { value, schemes, picking: false })
}

const getColor = (state, action) => {
  const { r, g, b, alpha } = action.payload
  // console.log(r)
  const schemes = schemesGenerator(r, g, b, state.schemeModel)
  const rgb = `rgb(${r}, ${g}, ${b})`,
    hex = rgbToHex(r, g, b)
  if (alpha !== 0 && state.picking) {
    return updatedObject(state, {
      rgbColors: { rgb, hex, r, g, b },
      schemes,
      alpha,
    })
  } else {
    return updatedObject(state, { alpha })
  }
}

const togglePick = state => {
  return updatedObject(state, { picking: !state.picking })
}

const setPickingToTrue = state => {
  return updatedObject(state, { picking: true })
}

const setPickingToFalse = state => {
  return updatedObject(state, { picking: false })
}

const schemeChange = (state, action) => {
  action.payload.event.persist()
  const { r, g, b } = state.rgbColors
  const { value } = action.payload.event.target
  const schemes = schemesGenerator(r, g, b, +value)
  return updatedObject(state, {
    schemeModel: +value,
    schemes,
  })
}

const inputValue = (state, action) => {
  const { value } = action.payload
  let rgbArr = [],
    rgb = '',
    hex = '',
    hsl = ''
  if (!COLOR_PATTERN.test(value)) {
    return setPickingToTrue(state)
  } else if (value.match(/^r/)) {
    rgbArr = value
      .replace(/[^\d,]/g, '')
      .split(',')
      .map(x => +x)
    rgb = value
    hex = rgbStringToHex(value)
  } else if (value.match(/^#/)) {
    rgbArr = hexToRgb(value)
    rgb = arrayToRgbString(rgbArr)
    hex = value
  } else if (value.match(/^h/)) {
    hsl = value
  }
  const [r, g, b] = rgbArr
  const schemes = schemesGenerator(r, g, b, state.schemeModel)
  return updatedObject(state, {
    rgbColors: { rgb, hex, r, g, b },
    schemes,
    hsl,
    picking: false,
  })
}

const reducer = (state = initialState, action) => {
  switch (action.type) {
    case actionTypes.VALUE_CONTROL:
      return valueControl(state, action)
    case actionTypes.GET_COLOR:
      return getColor(state, action)
    case actionTypes.TOGGLE_PICK:
      return togglePick(state)
    case actionTypes.SCHEME_CHANGE:
      return schemeChange(state, action)
    case actionTypes.INPUT_VALUE:
      return inputValue(state, action)
    case actionTypes.PICKING_TRUE:
      return setPickingToTrue(state, action)
    case actionTypes.PICKING_FALSE:
      return setPickingToFalse(state, action)
    default:
      return state
  }
}

export default reducer
