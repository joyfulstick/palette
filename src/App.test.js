import App from './App'
import { Provider } from 'react-redux'
import React from 'react'
import ReactDOM from 'react-dom'
import { createStore } from 'redux'
import reducer from './store/reducers/reducer'
import renderer from 'react-test-renderer'

describe('App', () => {
  const store = createStore(
    reducer,
    window.__REDUX_DEVTOOLS_EXTENSION__ &&
      window.__REDUX_DEVTOOLS_EXTENSION__(),
  )
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <Provider store={store}>
        <App />
      </Provider>,
      div,
    )
    ReactDOM.unmountComponentAtNode(div)
  })

  xit('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <App />
        </Provider>,
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
