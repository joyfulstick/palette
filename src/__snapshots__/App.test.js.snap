// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<div
  className="App"
>
  <main
    className="main"
  >
    <section
      className="hsv"
    >
      <canvas
        className="canvas--picked"
        height={342}
        onClick={[Function]}
        onMouseMove={[Function]}
        width={342}
      />
      <label
        className="range--value"
        htmlFor="range-value"
        style={
          Object {
            "width": 114,
          }
        }
      >
        <input
          className="range--value__input"
          in="range-value"
          max="1"
          min="0"
          onChange={[Function]}
          step=".01"
          type="range"
        />
        Value Control
      </label>
    </section>
    <article
      className="color-info"
    >
      <p>
        <span
          className="color-info__sample"
          onClick={[Function]}
          style={
            Object {
              "background": "rgb(72,160,255)",
              "color": "rgb(72,160,255)",
            }
          }
        >
          rgb
        </span>
        rgb(72,160,255)
      </p>
      <p>
        <span
          className="color-info__sample"
          onClick={[Function]}
          style={
            Object {
              "background": "#48a0ff",
              "color": "#48a0ff",
            }
          }
        >
          hex
        </span>
        #48a0ff
      </p>
    </article>
    <form
      className="schemes"
    >
      <label
        className="scheme__label"
        htmlFor="Monochromatic"
      >
        <input
          checked={false}
          id="Monochromatic"
          name="scheme"
          onChange={[Function]}
          type="radio"
          value={1}
        />
        Monochromatic
      </label>
      <label
        className="scheme__label"
        htmlFor="Complementary"
      >
        <input
          checked={true}
          id="Complementary"
          name="scheme"
          onChange={[Function]}
          type="radio"
          value={2}
        />
        Complementary
      </label>
      <label
        className="scheme__label"
        htmlFor="Triadic"
      >
        <input
          checked={false}
          id="Triadic"
          name="scheme"
          onChange={[Function]}
          type="radio"
          value={3}
        />
        Triadic
      </label>
      <label
        className="scheme__label"
        htmlFor="Analogous"
      >
        <input
          checked={false}
          id="Analogous"
          name="scheme"
          onChange={[Function]}
          type="radio"
          value={3.1}
        />
        Analogous
      </label>
      <label
        className="scheme__label"
        htmlFor="Compound"
      >
        <input
          checked={false}
          id="Compound"
          name="scheme"
          onChange={[Function]}
          type="radio"
          value={3.5}
        />
        Compound
      </label>
      <label
        className="scheme__label"
        htmlFor="Tetradic"
      >
        <input
          checked={false}
          id="Tetradic"
          name="scheme"
          onChange={[Function]}
          type="radio"
          value={4}
        />
        Tetradic
      </label>
    </form>
    <section
      className="swatches"
      onClick={[Function]}
    >
      <div
        className="swatches__samples"
      >
        <div
          className="samples__color"
          style={
            Object {
              "background": "#48a0ff",
            }
          }
        />
        <div
          className="samples__color"
          style={
            Object {
              "background": "#ffa748",
            }
          }
        />
      </div>
      <div
        className="swatches__samples"
      >
        <div
          className="samples__color"
          style={
            Object {
              "background": "#004fa3",
            }
          }
        />
        <div
          className="samples__color"
          style={
            Object {
              "background": "#a35500",
            }
          }
        />
      </div>
      <div
        className="swatches__samples"
      >
        <div
          className="samples__color"
          style={
            Object {
              "background": "#00346d",
            }
          }
        />
        <div
          className="samples__color"
          style={
            Object {
              "background": "#6d3900",
            }
          }
        />
      </div>
      <div
        className="swatches__samples"
      >
        <div
          className="samples__color"
          style={
            Object {
              "background": "#002752",
            }
          }
        />
        <div
          className="samples__color"
          style={
            Object {
              "background": "#522a00",
            }
          }
        />
      </div>
      <div
        className="swatches__samples"
      >
        <div
          className="samples__color"
          style={
            Object {
              "background": "#001f41",
            }
          }
        />
        <div
          className="samples__color"
          style={
            Object {
              "background": "#412200",
            }
          }
        />
      </div>
    </section>
    <button
      className="button--css"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "cursor": "pointer",
        }
      }
    >
      CSS
    </button>
    
    <label
      className="color-input"
      htmlFor="color-input"
    >
      <input
        className="color-input__element"
        name="color-input"
        onBlur={[Function]}
        onChange={[Function]}
        pattern="^(#[0-9a-f]{3}|#(?:[0-9a-f]{2}){2,4}|(rgb|hsl)a?\\\\((-?\\\\d+%?[,\\\\s]+){2,3}\\\\s*[\\\\d.]+%?\\\\))$"
        placeholder="HEX or RGB"
        type="text"
        value=""
      />
      Color Input
    </label>
  </main>
</div>
`;
